client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -765,3 +766,189 @@ export function* trashFile(system, path, id) {
+        if (app.version > latest.version) {
+          return app;
+        }
+        if (app.version < latest.version) {
+          return latest;
+        }
+        // Same version of app
+        if (app.revision >= latest.revision) {
+          return app;
+        }
+        return latest;
+      },
+      { revision: null, version: null }
+    );
+  return latestApp.id;
+};
+
+const getExtractParams = (file, latestExtract) => {
+  const inputFile = `agave://${file.system}${file.path}`;
+  const archivePath = `agave://${file.system}${file.path.substring(
+    0,
+    file.path.lastIndexOf('/') + 1
+  )}`;
+  return JSON.stringify({
+    allocation: 'FORK',
+    appId: latestExtract,
+    archive: true,
+    archivePath,
+    inputs: {
+      inputFile
+    },
+    maxRunTime: '02:00:00',
+    name: 'Extracting Compressed File',
+    parameters: {}
+  });
+};
+
+export const extractAppSelector = state => state.workbench.config.extractApp;
+
+export function* extractFiles(action) {
+  try {
+    const extractApp = yield select(extractAppSelector);
+    const latestExtract = yield call(getLatestApp, extractApp);
+    const params = getExtractParams(action.payload.file, latestExtract);
+    yield put({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'RUNNING', operation: 'extract' }
+    });
+    const submission = yield call(jobHelper, params);
+    if (submission.execSys) {
+      // If the execution system requires pushing keys, then
+      // bring up the modal and retry the extract action
+      yield put({
+        type: 'SYSTEMS_TOGGLE_MODAL',
+        payload: {
+          operation: 'pushKeys',
+          props: {
+            onSuccess: action,
+            system: submission.execSys,
+            onCancel: {
+              type: 'DATA_FILES_SET_OPERATION_STATUS',
+              payload: { status: 'ERROR', operation: 'extract' }
+            }
+          }
+        }
+      });
+    } else if (submission.status === 'ACCEPTED') {
+      yield put({
+        type: 'DATA_FILES_SET_OPERATION_STATUS',
+        payload: { status: 'SUCCESS', operation: 'extract' }
+      });
+    } else {
+      throw new Error('Unable to extract files');
+    }
+  } catch (error) {
+    yield put({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'ERROR', operation: 'extract' }
+    });
+  }
+}
+export function* watchExtract() {
+  yield takeLeading('DATA_FILES_EXTRACT', extractFiles);
+}
+
+/**
+ * Create JSON string of job params
+ * @async
+ * @param {Array<Object>} files
+ * @param {String} zipfileName
+ * @returns {String}
+ */
+const getCompressParams = (files, zipfileName, latestZippy) => {
+  const inputs = {
+    inputFiles: files.map(file => `agave://${file.system}${file.path}`)
+  };
+  const parameters = {
+    filenames: files.reduce((names, file) => `${names}"${file.name}" `, ''),
+    zipfileName,
+    compression_type: zipfileName.endsWith('.tar.gz') ? 'tgz' : '.zip'
```

Comment:
Instead of doing this check, we can pass the `fileType` value from the form [into the payload here](https://github.com/TACC/Core-Portal/pull/264/files#diff-5ded060fef1ebe8ad69ea1aa7bb4180715be1cf902760ee1bd5e7de1f20eaf63R72). If the value needs to be `tgz`, all we need to do is set the dropdown item's value to `tgz`, and the display value can stay as `.tar.gz`

Reply:
The `filetype` value is also used to set the filename. Could the check be moved to the component instead of within the saga?

Reply:
Ah good point. Yeah, that works too - and in that case this implementation in the saga also works. I'll resolve this 👍